### PR TITLE
BaSP-TG-135: implement firebase in server

### PR DESCRIPTION
**Pull request by JuliMonte5**
In this PR the following was done:
- Setup firebase service keys.
- Create middleware function.
- Add middleware to routes.
- Implemented firebase in Admins, Employees and Super Admins entities.

### CODE REVIEW

**Review by fede230298 — DISMISSED** [review]

**Review by aletiscornia — DISMISSED** [review]

**Review by MaicolFreire001 — APPROVED** [review]

File: src/controllers/employees.js [hunk 1]
```diff
@@ -53,19 +54,30 @@ const getEmployeeById = async (req, res) => {
...
       password: req.body.password,
+      firebaseUid: newFirebaseUser.uid,
     });
 
     const result = await employee.save();
     return res.status(201).json({
       message: 'Employee created successfully',
       data: result,
       error: false,
+      token: userToken,
```
Comment (ProDIGGY-vinilos) [inline src/controllers/employees.js:80]: plis delete it

File: src/controllers/admins.js [hunk 2]
```diff
@@ -89,12 +100,19 @@ const editAdmin = async (req, res) => {
...
     );
+
     if (!admin) {
       return res.status(404).json({
         message: `Admin with id:${req.params.id} not found`,
         error: true,
       });
     }
+
+    await firebase.auth().updateUser(admin.firebaseUid, {
```
Comment (ProDIGGY-vinilos) [inline src/controllers/admins.js:111]: change the admin.firebaseUid to req.body.firebaseUid

File: src/controllers/employees.js (hunk 1 above)
Comment (JuliMonte5) [reply src/controllers/employees.js:80]: Done

File: src/controllers/admins.js (hunk 2 above)
Comment (JuliMonte5) [reply src/controllers/admins.js:111]: Done

**Review by ProDIGGY-vinilos — APPROVED** [review]

**Pull request merged**